doc/compiler/il/Node.md

```diff
@@ -0,0 +1,91 @@
+# Node
```

Comment:
* General comment:  use the capitalized `Node` rather than `node` throughout this text.  I caught some of these, but not all.